### PR TITLE
Feat: Add BitBucketCredentials block

### LINKED ISSUE #1: Create `BitbucketCredentials` block
Create a block capable of holding the credentials necessary to clone a Bitbucket repository. 

The block should have fields for a [HTTP access token](https://confluence.atlassian.com/bitbucketserver/http-access-tokens-939515499.html) and an optional username. 

Project and repository tokens do not require a username while use tokens do require a username. This block should be able to support both usages. 

The token field should be a `SecretStr` in order to protect the token from exposure.

### PR DESCRIPTION
**Pull request by serinamarie**
Closes #1 

### Checklist

- [x] References any related issue by including "Closes #<Issue Number>" or "Closes <Issue URL>".
  - If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect-bitbucket/issues/new/choose) first.
- [x] Includes tests or only affects documentation.
- [x] Passes `pre-commit` checks.
  - Run `pre-commit install && pre-commit run --all` locally for formatting and linting.
- [x] Includes screenshots of documentation updates.
  - Run `mkdocs serve` view documentation locally.
- [x] Summarizes PR's changes in [CHANGELOG.md](https://github.com/PrefectHQ/prefect-bitbucket/blob/main/CHANGELOG.md)

Screen Shot 2022-11-21 at 10 11 13 PM

Screen Shot 2022-11-21 at 10 00 54 PM

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ## Unreleased
 
 ### Added
+- `BitBucketCredentials` block - [#1](https://github.com/PrefectHQ/prefect-bitbucket)
```
Comment (ahuang11) [inline CHANGELOG.md]:
```suggestion
- `BitBucketCredentials` block - [#3](https://github.com/PrefectHQ/prefect-bitbucket/pull/3/)
```

File: README.md [hunk 2]
```diff
@@ -1,7 +1,5 @@
 # prefect-bitbucket
 
-## 🚧 UNDER CONSTRUCTION 🚧
```
Comment (ahuang11) [inline README.md]: I think we still want to keep this until release

File: README.md [hunk 3]
```diff
@@ -50,21 +48,36 @@ prefect block register -m prefect_bitbucket.credentials
 Note, to use the `load` method on Blocks, you must already have a block document [saved through code](https://orion-docs.prefect.io/concepts/blocks/#saving-blocks) or [saved through the UI](https://orion-docs.prefect.io/ui/blocks/).
 
 ### Write and run a flow
+Load a pre-existing BitBucketCredentials block
```
Comment (ahuang11) [inline README.md]:
```suggestion
#### Load a pre-existing BitBucketCredentials block.
```

File: README.md [hunk 4]
```diff
@@ -50,21 +48,36 @@ prefect block register -m prefect_bitbucket.credentials
...
 from prefect import flow
-from prefect_bitbucket.tasks import (
-    goodbye_prefect_bitbucket,
-    hello_prefect_bitbucket,
-)
+from prefect_bitbucket.credentials import BitBucketCredentials
+
+@flow
+def use_stored_bitbucket_creds_flow():
+    """Create a flow"""
```
Comment (ahuang11) [inline README.md]:
```suggestion
```

File: README.md [hunk 5]
```diff
@@ -50,21 +48,36 @@ prefect block register -m prefect_bitbucket.credentials
...
+    bitbucket_credentials_block = BitBucketCredentials.load("BLOCK_NAME")
 
+    return bitbucket_credentials_block
+
+use_stored_bitbucket_creds_flow()
+```
+
+Create a new BitBucketCredentials block in a flow
```
Comment (ahuang11) [inline README.md]:
```suggestion
#### Create a new BitBucketCredentials block in a flow.
```

File: prefect_bitbucket/credentials.py [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+    username: str = None
+
+    @validator("username")
+    def _validate_username(cls, value: str) -> str:
+        """Validators are by default only called on provided arguments."""
+        pattern = "^[A-Za-z0-9_-]*$"
+        assert bool(
+            re.match(pattern, value)
+        ), "username must be alpha, num, dash and/or underscore only"
+        assert len(value) <= 30, "username cannot be longer than 30 chars"
```
Comment (ahuang11) [inline prefect_bitbucket/credentials.py]: I'm hesitant to use assert in validations; I prefer `if ...: raise ValueError(msg)`
https://stefan.sofa-rockers.org/2018/04/16/assertions-and-exceptions/

File: prefect_bitbucket/credentials.py [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+        Load stored BitBucket credentials:
+        ```python
+        from prefect_bitbucket import BitBucketCredentials
+        bitbucket_credentials_block = BitBucketCredentials.load("BLOCK_NAME")
+        ```
+    """
+
+    _block_type_name = "BitBucket Credentials"
+    _logo_url = HttpUrl(
+        url="",  # REPLACE
```
Comment (ahuang11) [inline prefect_bitbucket/credentials.py]: A review comment as placeholder

**Review by ahuang11 — CHANGES_REQUESTED** [review]
Great start! Some suggestions.

Also can you help add tests and the logo URL?

Lastly, I was wondering if you're interested in testing out pydocstyle for this collection? https://github.com/PrefectHQ/prefect-collection-template/issues/107
https://github.com/PrefectHQ/prefect-collection-template/issues/107

File: prefect_bitbucket/credentials.py [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+        from prefect_bitbucket import BitBucketCredentials
+        bitbucket_credentials_block = BitBucketCredentials.load("BLOCK_NAME")
+        ```
+    """
+
+    _block_type_name = "BitBucket Credentials"
+    _logo_url = HttpUrl(
+        url="https://banner2.cleanpng.com/20180320/gcq/kisspng-symbol-font-bitbucket-5ab0b6a9021793.8467511015215305370086.jpg",  # noqa
+        scheme="https",
+    )
```
Comment (ahuang11) [inline prefect_bitbucket/credentials.py]:
```suggestion
    _logo_url = "https://images.ctfassets.net/gm98wzqotmnx/27LMR24ewTSDW238Lks1vH/34c5028659f4007528feadc8db8cecbd/500px-Bitbucket-blue-logomark-only.svg.png?h=250"  # noqa
```

File: prefect_bitbucket/credentials.py [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+    token: SecretStr = Field(
+        name="Personal Access Token",
+        default=None,
```
Comment (ahuang11) [inline prefect_bitbucket/credentials.py:30]: This should be required right?
```suggestion
        default=...,
```

File: prefect_bitbucket/credentials.py [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+        description="A BitBucket Personal Access Token.",
+    )
+    username: str = None
```
Comment (ahuang11) [inline prefect_bitbucket/credentials.py]: Not sure if this should be optional or required:
```suggestion
    username: Optional[str] = Field(default=None, description="Identification name unique across entire BitBucket site.")
```

**Review by ahuang11 — CHANGES_REQUESTED** [review]
Just a couple more suggestions. Thanks!

File: prefect_bitbucket/credentials.py (hunk 10 above)
Comment (serinamarie) [reply prefect_bitbucket/credentials.py]: @ahuang11 Alex specified it to be optional

Comment (ahuang11) [reply prefect_bitbucket/credentials.py]: Okay, I updated my suggestion above.

**Comment by serinamarie** [comment]
> Lastly, I was wondering if you're interested in testing out pydocstyle for this collection? [PrefectHQ/prefect-collection-template#107](https://github.com/PrefectHQ/prefect-collection-template/issues/107) [PrefectHQ/prefect-collection-template#107](https://github.com/PrefectHQ/prefect-collection-template/issues/107)

We can try it for the repository block PR if that sounds fine with you

**Comment by ahuang11** [comment]
Sure

**Review by ahuang11 — APPROVED** [review]
Looks good!

**Pull request merged**